BASE/src/Microsoft.ApplicationInsights/Channel/TaskEx.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace Microsoft.ApplicationInsights.Channel
```

Comment:
I see that you want to use this in multiple projects.
I recommend moving this to the "Common.Base" project at "BASE\src\Common\Common\".
You could do this as a separate PR.

Reply:
We could completely remove common project with internals visible, working this on a separate PR.